### ISSUE TITLE
[registry api incompatibility] older nginx (v1 image schema) example error

### ISSUE THREAD

**Issue by jpmcb** [k14s/kbld#21]
Howdy! I am following the example found on https://get-kbld.io/ with deploying a simple nginx server to a new GKE cluster. I've used the following command with the yaml provided in the example:

```
$ kbld -f k8s/ | kubectl apply -f -
resolve | final: nginx:1.7.9 -> index.docker.io/library/nginx@sha256:03bf9c90c36067a6d328b184f6c6068766fa5c60681adcced2509ae85c14b983
deployment.apps/nginx-deployment created
```

 And i'm receiving the following error on the nginx pods:

```
Warning  Failed     3m2s (x4 over 4m33s)   kubelet, gke-johns-test-default-pool-9c3cb85a-bzrj  Failed to pull image "index.docker.io/library/nginx@sha256:03bf9c90c36067a6d328b184f6c6068766fa5c60681adcced2509ae85c14b983": rpc error: code = Unknown desc = Error response from daemon: manifest for nginx@sha256:03bf9c90c36067a6d328b184f6c6068766fa5c60681adcced2509ae85c14b983 not found
```

When attempting to pull the nginx image locally:
```
$ docker pull nginx:1.7.9
1.7.9: Pulling from library/nginx
Image docker.io/library/nginx:1.7.9 uses outdated schema1 manifest format. Please upgrade to a schema2 image for better future compatibility. More information at https://docs.docker.com/registry/spec/deprecated-schema-v1/
a3ed95caeb02: Pull complete 
6f5424ebd796: Pull complete 
d15444df170a: Pull complete 
e83f073daa67: Pull complete 
a4d93e421023: Pull complete 
084adbca2647: Pull complete 
c9cec474c523: Pull complete 
Digest: sha256:e3456c851a152494c3e4ff5fcc26f240206abac0c9d794affb40e0714846c451
Status: Downloaded newer image for nginx:1.7.9
docker.io/library/nginx:1.7.9
```

Could this be related to the old schema manifest not working with k8s? I also note that the digest I get on the local docker pull is different from the resolve I get on the `kbld` command. 

Changing the image line in the yaml
```
image: nginx:latest
```
is successful.

Let me know if there are questions or if I can provide more information!

**`kbld`**:
v0.13.0

**Kubernetes**:
Client - v1.15.3
Server - v1.14.10-gke.17

**Docker**:
macOS desktop - 2.2.0.3
Engine - 19.03.5

**Comment by jpmcb** [comment]
Also of note, when performing the following command:
```
watch kbld -f k8s/
```
where the file for the nginx deployment is:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80
```
The sha digets _changes_ upon every iteration of watch. For example, on first iteration
```
resolve | final: nginx:1.7.9 -> index.docker.io/library/nginx@sha256:2d190f022cf883acd41f475f461b1e28320ebe826a9358b4d3f98fab8c9b8af3
```
and on second iteration
```
resolve | final: nginx:1.7.9 -> index.docker.io/library/nginx@sha256:25931ac2fa92c76f5b4e5465cae87c8d87c8f201e6ace9e0427c53b8da65c35d
```

**Comment by cppforlife** [comment]
that's wild that it's returning different digest every time instead of failing but yeah it is related to image being v1 schema [1]. im guessing v1 schema _did not_ have a "stable" digest that was being returned, so api is doing some magic.

as you said later versions of nginx (that do not use v1 schema) are ok. (e.g. `echo "image: nginx:1.17.8" | kbld -f-` results in consistent digest)

in general kbld is built on go-containerregistry library and does not support v1 schema images.

i wonder if it's worth for kbld to resolve each image twice to make sure that digest is being returned correctly, and fail if it's not.

[1] `Image docker.io/library/nginx:1.7.9 uses outdated schema1 manifest format. Please upgrade to a schema2 image for better future compatibility. More information at https://docs.docker.com/registry/spec/deprecated-schema-v1/`

**Comment by jpmcb** [comment]
Wow, very interesting. Yes, I agree, if the digest is not consistent, it would make sense to fail since the API is maybe trying to do it's best. Thanks for looking into this!

**Comment by cppforlife** [comment]
ImageOverride is your friend (https://github.com/k14s/kbld/blob/develop/docs/config.md#imageoverrides) if you want to preresolve the image to whatever you want (ie have kbld ignore it).

**Comment by cppforlife** [comment]
made kbld double check returned digest: https://github.com/k14s/kbld/commit/91a5653015367ce23f5bcc1089ed1af9bfba8c0c. available in 0.14.0.
